.github/workflows/main.yml

```diff
@@ -186,3 +196,78 @@ jobs:
         name: taisei_windows_fail_log
         path: build/meson-logs/meson-log.txt
         if-no-files-found: warn
+
+  emscripten-test-build:
+    name: Emscripten Test Build (WebGL)
+    if: "!contains(github.event.head_commit.message, '[skip ci]')"
+    runs-on: ubuntu-20.04
+    steps:
+    - name: Checkout Code
+      uses: actions/checkout@v2
+      with:
+        submodules: 'recursive'
+
+    - name: Set Up Build Environment
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+
+    - name: Install Build Tools
+      run: >
+        sudo apt update
+
+        sudo apt install -y -qq
+        python3-docutils
+        python3-pip
+        git
+
+        pip install
+        meson==0.56.2
+        ninja
+        zstandard
+
+    - name: Pull cached emsdk
+      id: cache-emsdk
+      uses: actions/cache@v2
+      with:
+        path: ${{ env.EM_CACHE_FOLDER }}
+        key: ${{ env.EM_VERSION }}-${{ runner.os }}
+
+    - name: Install emsdk
+      if: steps.cache-emsdk.outputs.cache-hit != 'true'
+      run: |
+        rm -rf emsdk/
+        git clone https://github.com/emscripten-core/emsdk.git
+        emsdk/emsdk install ${{ env.EM_VERSION }}
+        emsdk/emsdk activate ${{ env.EM_VERSION }}
+
+    - name: Verify emsdk
+      run: |
+        source emsdk/emsdk_env.sh
+        emcc -v
+
+    - name: Configure (Meson)
+      run: >
+        source emsdk/emsdk_env.sh
+
+        meson setup build/
+        --cross-file misc/ci/emscripten-build-test-ci.ini
+        --prefix=$(pwd)/build-test
+
+        meson configure
+        -Dbuild.cpp_std=gnu++14
+        build/
+
+    - name: Build
+      run: |
+        source emsdk/emsdk_env.sh
+        # run twice due to a header race condition with SPIRV-Tools
+        meson compile txz -C build/ --verbose || true
+        meson compile txz -C build/ --verbose
```

Comment:
Can you elaborate on what's going on here? Sounds like something we definitely want fixed.

Reply:
Side note, might be better to use `tar` instead of `txz` here. The `xz` compression is quote slow and doesn't reduce the file size much here (and we ignore the artifact, anyway).

Reply:
For the error, here's the log: https://github.com/StarWitch/taisei/runs/2522851326?check_suite_focus=true

```
 FAILED: subprojects/SPIRV-Tools/source/opt/libSPIRV-Tools-opt-native.a.p/pch.h.pch 
c++ -Isubprojects/SPIRV-Tools/source/opt/libSPIRV-Tools-opt-native.a.p -Isubprojects/SPIRV-Tools/source/opt -I../subprojects/SPIRV-Tools/source/opt -Isubprojects/SPIRV-Tools -I../subprojects/SPIRV-Tools -Isubprojects/SPIRV-Tools/source -I../subprojects/SPIRV-Tools/source -I../subprojects/SPIRV-Tools/include -I../subprojects/SPIRV-Headers/include -Xclang -fcolor-diagnostics -DNDEBUG -pipe -Wall -Winvalid-pch -Wnon-virtual-dtor -std=gnu++14 -x c++-header -MD -MQ subprojects/SPIRV-Tools/source/opt/libSPIRV-Tools-opt-native.a.p/pch.h.pch -MF subprojects/SPIRV-Tools/source/opt/libSPIRV-Tools-opt-native.a.p/pch.h.pch.d -o subprojects/SPIRV-Tools/source/opt/libSPIRV-Tools-opt-native.a.p/pch.h.pch -c ../subprojects/SPIRV-Tools/source/opt/meson-pch/pch.h
In file included from ../subprojects/SPIRV-Tools/source/opt/meson-pch/pch.h:24:
In file included from ../subprojects/SPIRV-Tools/source/opt/basic_block.h:29:
../subprojects/SPIRV-Tools/source/opt/instruction.h:25:10: fatal error: 'OpenCLDebugInfo100.h' file not found
#include "OpenCLDebugInfo100.h"
         ^~~~~~~~~~~~~~~~~~~~~~
```

Basically, the build gets ahead of itself, and can occasionally fail to find this file because it isn't in place yet. Running the command again fixes it.

Reply:
Seems like SPIRV-Tools's internal dependencies aren't set up adequately. I'll look into this soon.

Reply:
Cool. If you don't have time to investigate/fix it right now, this workaround does work for now. We can always remove it when you commit that change.